### PR TITLE
Prompt 1password to save Secret Key

### PR DESCRIPTION
**Pull request by kyranjamie**
1password-popup

### CODE REVIEW

File: src/components/onboarding/screens/secret-key.tsx [hunk 1]
```diff
@@ -32,7 +32,23 @@ export const SecretKey: React.FC<SecretKeyProps> = props => {
   });
 
   return (
-    <>
+    <form
```
Comment (hstove) [inline src/components/onboarding/screens/secret-key.tsx]: Thomas recently added an `onSubmit` prop to screen, see pages like `Username` for an example. It'll generate a `<form>` for you.

**Comment by hstove** [comment]
What are you thoughts about how this _could_ work?

**Review by aulneau — APPROVED** [review]

File: src/components/onboarding/screens/username.tsx [hunk 2]
```diff
@@ -44,15 +46,21 @@ interface UsernameProps {
...
 
-  const { wallet, screen } = useSelector((state: AppState) => ({
+  const { wallet, screen, secretKey, appName } = useSelector((state: AppState) => ({
     wallet: selectCurrentWallet(state) as Wallet,
     screen: selectCurrentScreen(state),
+    secretKey: selectSecretKey(state),
+    appName: selectAppName(state),
   }));
 
+  const title = `${appName} with Blockstack`;
```
Comment (hstove) [inline src/components/onboarding/screens/username.tsx]: I definitely appreciate the intention here, but changing the actual screen title makes this page much more confusing. It went from "Choose a Username" to "AppName with Blockstack". Did you only mean to update the `document.title`? There must be a way to use a hidden label, or something, to do this without updating the page's copy.

image

**Review by hstove — CHANGES_REQUESTED** [review]
Overall this is good, but I think we definitely need to have a better solution around the screen's title being changed

File: src/components/onboarding/screens/username.tsx (hunk 2 above)
Comment (kyranjamie) [reply src/components/onboarding/screens/username.tsx]: Open to suggestions, I'd love another way of doing it. This less-than-ideal title hack is better than having a password entry saved as "Choose an account", no? [Asked @1password on Twitter](https://twitter.com/kyranjamie/status/1231936609240649728?s=20)

Comment (kyranjamie) [reply src/components/onboarding/screens/username.tsx]: I also tried changing on submit, but it has to happen before then

**Review by hstove — APPROVED** [review]

**Comment by timstackblock** [comment]
This is not working as expected on the hosted app

I can copy the secret but it does not trigger 1password

**Pull request closed without merging**